kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala

```diff
@@ -154,14 +167,25 @@ class OperationLog(path: Path) {
     tRow
   }
 
+  def read(maxRows: Int): TRowSet = synchronized {
+    read(FETCH_NEXT, maxRows)
+  }
+
   /**
    * Read to log file line by line
    *
    * @param maxRows maximum result number can reach
+   * @param order   the  fetch orientation of the result, can be FETCH_NEXT, FETCH_FIRST
    */
-  def read(maxRows: Int): TRowSet = synchronized {
+  def read(order: FetchOrientation = FETCH_NEXT, maxRows: Int): TRowSet = synchronized {
     if (!initialized) return ThriftUtils.newEmptyRowSet
-    val (logs, lines) = readLogs(reader, maxRows, maxRows)
+    if (order != FETCH_NEXT && order != FETCH_FIRST) {
+      throw KyuubiSQLException(s"$order in operation log is not supported")
+    }
+    if (order == FETCH_FIRST) {
+      resetReader()
```

Comment:
how about just close seekableReader and create a new seekableReader?

Reply:
I think just closing the ‘seekableReader’ and creating a new ‘seekableReader’ may not work because ‘seekableReader’ and ‘reader’ are different objects. Unless all instances of ‘reader’ are also replaced with ‘seekableReader’,